finality-aleph/src/new_network/service.rs

```diff
@@ -165,22 +171,425 @@ impl<N: Network, D: Data> Service<N, D> {
-                    }
-                },
+                Some(()) =  Self::send(&self.network, &mut self.to_send) => { },
             }
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{ConnectionCommand, DataCommand, Service, IO};
+    use crate::new_network::{
+        manager::testing::MockNetworkIdentity, mock::MockNetwork, NetworkIdentity, Protocol,
+        ALEPH_PROTOCOL_NAME, ALEPH_VALIDATOR_PROTOCOL_NAME,
+    };
+    use codec::Encode;
+    use futures::{
+        channel::{mpsc, oneshot},
+        Future, StreamExt,
+    };
+    use sc_network::{
+        multiaddr::Protocol as ScProtocol, Event, Multiaddr as ScMultiaddr, ObservedRole,
+    };
+    use std::iter::FromIterator;
+    use std::{borrow::Cow, collections::HashSet, iter};
+    use substrate_test_runtime_client::runtime::Block;
+
+    type MockData = Vec<u8>;
+
+    pub struct MockIO {
+        messages_for_user: mpsc::UnboundedSender<(MockData, DataCommand)>,
+        messages_from_user: mpsc::UnboundedReceiver<MockData>,
+        commands_for_manager: mpsc::UnboundedSender<ConnectionCommand>,
+    }
+
+    async fn prepare() -> (
+        impl Future<Output = sc_service::Result<(), tokio::task::JoinError>>,
+        oneshot::Sender<()>,
+        MockNetwork<Block>,
+        MockIO,
+    ) {
+        let (mock_messages_for_user, messages_from_user) = mpsc::unbounded();
+        let (messages_for_user, mock_messages_from_user) = mpsc::unbounded();
+        let (mock_commands_for_manager, commands_from_manager) = mpsc::unbounded();
+        let io = IO {
+            messages_from_user,
+            messages_for_user,
+            commands_from_manager,
+        };
+        let mock_io = MockIO {
+            messages_for_user: mock_messages_for_user,
+            messages_from_user: mock_messages_from_user,
+            commands_for_manager: mock_commands_for_manager,
+        };
+
+        let (oneshot_tx, event_stream_taken_oneshot_rx) = oneshot::channel();
+        let network = MockNetwork::new(oneshot_tx);
+        let service = Service::new(network.clone(), io);
+
+        let (exit_tx, exit_rx) = oneshot::channel();
+        let task_handle = async move {
+            tokio::select! {
+                _ = service.run() => { },
+                _ = exit_rx => { },
+            };
+        };
+        let service_handle = tokio::spawn(task_handle);
+
+        // wait till service takes the event_stream
+        event_stream_taken_oneshot_rx.await.unwrap();
+
+        (service_handle, exit_tx, network, mock_io)
+    }
+
+    #[tokio::test]
+    async fn test_sync_connected() {
+        let (service_handle, exit_tx, mut network, _mock_io) = prepare().await;
+
+        let identity = MockNetworkIdentity::new().identity();
+
+        network.emit_event(Event::SyncConnected {
+            remote: identity.1.into(),
+        });
+
+        let expected = (
+            iter::once(ScMultiaddr::empty().with(ScProtocol::P2p(identity.1 .0.into()))).collect(),
```

Comment:
```suggestion
            iter::once(ScMultiaddr::empty().with(ScProtocol::P2p(identity.1.0.into()))).collect(),
```

Reply:
Well, the formatter changes it to be like that :(

Reply:
wow :O